development/tools/pkg/pjtester/pjtester.go

```diff
@@ -100,333 +133,298 @@ func checkEnvVars(varsList []string) error {
 	return nil
 }
 
-// newProwK8sClientset is building Prow client for kubeconfig location provided as env variable
-func newProwK8sClientset() *prowclient.Clientset {
+// newProwK8sClientset is building Prow client for provided kubeconfig.
+// Kubeconfig location is taken from KUBECONFIG_PATH env variable.
+func newProwK8sClientset() (*prowclient.Clientset, error) {
 	k8sConfig, err := clientcmd.BuildConfigFromFlags("", os.Getenv("KUBECONFIG_PATH"))
 	if err != nil {
-		log.WithError(err).Fatalf("Failed create config for prow k8s clientset.")
+		return nil, fmt.Errorf("failed create config for prow k8s clientset")
 	}
 	clientset, err := prowclient.NewForConfig(k8sConfig)
 	if err != nil {
-		log.WithError(err).Fatalf("Failed create prow k8s clientset.")
+		return nil, fmt.Errorf("failed create prow k8s clientset")
 	}
-	return clientset
+	return clientset, nil
 }
 
 // readTestCfg read and validate data from pjtester.yaml file.
-// It will set default path for prowjobs and config files if not provided in a file.
-func readTestCfg(testCfgFile string) testCfg {
+// Validation is done based on requirements defined in struct fields tags.
+func readTestCfg(testCfgFile string) (testCfg, error) {
 	var t testCfg
 	yamlFile, err := os.ReadFile(testCfgFile)
 	if err != nil {
-		log.Fatal("Failed read test config file from virtual path KYMA_PROJECT_DIR/test-infra/vpath/pjtester.yaml")
+		return testCfg{}, fmt.Errorf("failed read test config file from vpath/pjtester.yaml")
 	}
 	err = yaml.Unmarshal(yamlFile, &t)
 	if err != nil {
-		log.Fatal("Failed unmarshal test config yaml.")
-	}
-	if len(t.PjNames) == 0 {
-		log.WithError(err).Fatalf("PjNames were not provided.")
-	}
-	for index, pj := range t.PjNames {
-		if pj.PjName == "" {
-			log.WithError(err).Fatalf("jobName to test was not provided.")
-		}
-		if pj.PjPath == "" {
-			t.PjNames[index].PjPath = defaultPjPath
-		}
-	}
-	if t.ConfigPath == "" {
-		t.ConfigPath = defaultConfigPath
+		return testCfg{}, fmt.Errorf("failed unmarshal test config yaml")
 	}
-	if len(t.PrConfigs) > 0 {
-		for _, repo := range t.PrConfigs {
-			if len(repo) > 0 {
-				for _, pr := range repo {
-					if pr.PrNumber == 0 {
-						log.WithError(err).Fatalf("Pull request number for repo was not provided.")
-					}
-				}
-			} else {
-				log.WithError(err).Fatalf("Pull request number for repo was not provided.")
-			}
+
+	validate := validator.New()
+	// returns nil or ValidationErrors ( []FieldError )
+	err = validate.Struct(t)
+	if err != nil {
+		// this check is only needed when your code could produce
+		// an invalid value for validation such as interface with nil
+		// value most including myself do not usually have code like this.
+		if _, ok := err.(*validator.InvalidValidationError); ok {
+			fmt.Println(err)
+			return testCfg{}, err
 		}
+		// from here you can create your own error messages in whatever language you wish
+		return testCfg{}, err
```

Comment:
Do we need to differentiate between error types just to Print error in this function? Shouldn't we just return err no matter the type if we don't do any operations on it based on type?

Reply:
Not needed, removed.